### PR TITLE
Create a true many-to-many relationship for Groups and ContentItems

### PR DESCRIPTION
**Pull request by whoojemaflip**
By creating a join table we make it faster to query larger groups. 

This migration assumes there are no Groups as there are none in my recently replicated data. 
*It will totally squash any data that's in the Groups->content_item_ids field*

### CODE REVIEW

File: app/models/importers/pagelist_for_concept_mining.rb [hunk 1]
```diff
@@ -0,0 +1 @@
+pagelist_for_concept_mining.rb
```
Comment (pmanrubia) [inline app/models/importers/pagelist_for_concept_mining.rb]: Do we need this importer? It seems unrelated to this PR

Comment (whoojemaflip) [inline app/models/importers/pagelist_for_concept_mining.rb]: ah good catch. thought I had stashed that.

**Comment by whoojemaflip** [comment]
@pmanrubia done
